src/core_functions/scalar/date/date_part.cpp

```diff
@@ -891,11 +894,15 @@ int64_t DatePart::YearWeekOperator::Operation(dtime_t input) {
 
 template <>
 int64_t DatePart::EpochNanosecondsOperator::Operation(timestamp_t input) {
+	if (!Timestamp::IsFinite(input)) {
```

Comment:
The `DatePart::UnaryFunction` wrapper already checks this and returns NULL, so I think that is the correct behaviour. Just an assertion is fine.